code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPPushTemplate.java

```diff
@@ -0,0 +1,409 @@
+        try {
+            final JSONArray jsonArray = new JSONArray(actionButtons);
+            for (int i = 0; i < jsonArray.length(); i++) {
+                final JSONObject jsonObject = jsonArray.getJSONObject(i);
+                final ActionButton button = getActionButton(jsonObject);
+                if (button == null) continue;
+                actionButtonList.add(button);
+            }
+        } catch (final JSONException e) {
+            Log.warning(CampaignPushConstants.LOG_TAG, SELF_TAG, "Exception in converting actionButtons json string to json object, Error : %s", e.getLocalizedMessage());
+            return null;
+        }
+        return actionButtonList;
+    }
+
+    private ActionButton getActionButton(final JSONObject jsonObject) {
+        try {
+            final String label = jsonObject.getString(ActionButtons.LABEL);
+            if (label.isEmpty()) {
+                Log.debug(CampaignPushConstants.LOG_TAG, SELF_TAG, "Label is empty");
+                return null;
+            }
+            String uri = null;
+            final String type = jsonObject.getString(ActionButtons.TYPE);
+            if (type.equals(ActionButtonType.WEBURL) || type.equals(ActionButtonType.DEEPLINK)) {
+                uri = jsonObject.optString(ActionButtons.URI);
+            }
+
+            Log.trace(CampaignPushConstants.LOG_TAG, SELF_TAG, "Creating an ActionButton with label (%s), uri (%s), and type (%s)", label, uri, type);
+            return new ActionButton(label, uri, type);
+        } catch (final JSONException e) {
+            Log.warning(CampaignPushConstants.LOG_TAG, SELF_TAG, "Exception in converting actionButtons json string to json object, Error : %s", e.getLocalizedMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Enum to denote the type of action
+     */
+    public enum ActionType {
+        DEEPLINK, WEBURL, DISMISS, OPENAPP, NONE
+    }
+
+    /**
+     * Class representing the action button with label, link and type
+     */
+    public class ActionButton {
+        private final String label;
+        private final String link;
+        private final ActionType type;
+
+
+        public ActionButton(final String label, final String link, final String type) {
+            this.label = label;
+            this.link = link;
+            this.type = getActionTypeFromString(type);
+        }
+
+        public String getLabel() {
+            return label;
+        }
+
+        public String getLink() {
+            return link;
+        }
+
+        public ActionType getType() {
+            return type;
+        }
+    }
+
+    private int getNotificationPriorityFromString(final String priority) {
+        if (priority == null) return Notification.PRIORITY_DEFAULT;
+        final Integer resolvedPriority = notificationPriorityMap.get(priority);
+        if (resolvedPriority == null) return Notification.PRIORITY_DEFAULT;
+        return resolvedPriority;
+    }
```

Comment:
Can you move this to `NotificationPriority` as  `static int NotificationPriority.from(final String priority)`